packages/expo-modules-autolinking/scripts/ios/project_integrator.rb

```diff
@@ -128,5 +132,124 @@ def self.set_autolinking_configuration(project)
+        generate_support_script(autolinking_manager, modules_provider_path)
+      )
+
+      # Make the support script executable
+      FileUtils.chmod('+x', support_script_path)
+
+      # Force the build phase script to run on each build (including incremental builds)
+      xcode_build_script.always_out_of_date = '1'
+
+      # Make sure the build script in Xcode is up to date, but probably it's not going to change
+      # as it just runs the script generated in the target support files
+      xcode_build_script.shell_script = generate_xcode_build_script(support_script_relative_path)
+    end
+
+    # Generates the shell script of the build script phase.
+    # Try not to modify this since it involves changes in the pbxproj so
+    # it's better to modify the support script instead, if possible.
+    def self.generate_xcode_build_script(script_relative_path)
+      escaped_path = script_relative_path.to_s.gsub(/[^a-zA-Z0-9,\._\+@%\/\-]/) { |char| "\\#{char}" }
+
+      <<~XCODE_BUILD_SCRIPT
+      # This script configures Expo modules and generates the modules provider file.
+      bash -l -c "./#{escaped_path}"
+      XCODE_BUILD_SCRIPT
+    end
+
+    # Generates the support script that is executed by the build script phase.
+    def self.generate_support_script(autolinking_manager, modules_provider_path)
+      args = autolinking_manager.base_command_args.map { |arg| "\"#{arg}\"" }.join(' ')
+
+      <<~SUPPORT_SCRIPT
+      #!/usr/bin/env bash
+      # @generated by expo-modules-autolinking
+
+      set -eo pipefail
+
+      function node() {
+        # Start with a default
+        NODE_BINARY=$(command -v node)
+        export NODE_BINARY
+
+        # Override the default with the global environment
+        ENV_PATH="$PODS_ROOT/../.xcode.env"
+        if [[ -f "$ENV_PATH" ]]; then
+          source "$ENV_PATH"
+        fi
+
+        # Override the global with the local environment
+        LOCAL_ENV_PATH="${ENV_PATH}.local"
+        if [[ -f "$LOCAL_ENV_PATH" ]]; then
+          source "$LOCAL_ENV_PATH"
+        fi
+
+        if [[ -n "$NODE_BINARY" && -x "$NODE_BINARY" ]]; then
+          echo "Node found at: ${NODE_BINARY}"
+        else
+          cat >&2 << NODE_NOT_FOUND
+      error: Could not find "node" executable while running an Xcode build script.
+      You need to specify the path to your Node.js executable by defining an environment variable named NODE_BINARY in your project's .xcode.env or .xcode.env.local file.
+      You can set this up quickly by running:
+
+      echo "export NODE_BINARY=\\$(command -v node)" >> .xcode.env
+
+      in the ios folder of your project.
+      NODE_NOT_FOUND
+
+          exit 1
+        fi
+
+        # Execute argument, if present
+        if [[ "$#" -gt 0 ]]; then
+          "$NODE_BINARY" "$@"
+        fi
+      }
+
+      node --eval "require(\'expo-modules-autolinking\')(process.argv.slice(1))" generate-package-list #{args} --target "#{modules_provider_path}"
```

Comment:
```suggestion
      with_node --eval "require(\'expo-modules-autolinking\')(process.argv.slice(1))" generate-package-list #{args} --target "#{modules_provider_path}"
```

rename to `with_node` maybe. it's confusing whether it's a direct `node` command execution until i find the function above is also renaming as `node`